pkg/kclient/deployments.go

```diff
@@ -364,3 +364,31 @@ func (c *Client) GetDeploymentConfigsFromSelector(selector string) ([]appsv1.Dep
 	}
 	return dcList.Items, nil
 }
+
+// GetDeploymentAPIVersion returns a map with Group, Version, Resource information of Deployment objects
+// depending on the GVR supported by the cluster
+func (c *Client) GetDeploymentAPIVersion() (map[string]string, error) {
```

Comment:
Would it be better to use `metav1.GroupVersionResource` as a return type, instead of `map`?

Reply:
Makes sense to me. Pushed the change. Haven't tested it, however.